inference-engine/src/transformations/include/transformations/low_precision/layer_transformation.hpp

```diff
@@ -0,0 +1,380 @@
+
+    void setParamsManager(IParamsManager* paramsManager) noexcept;
+    void setLayerTransformationsManager(ILayerTransformationsManager* layerTransformationsManager) noexcept;
+
+    void setUpdatePrecisions(const bool updatePrecisions);
+    void setQuantizedTensorAlignmentOnActivations(const QuantizedTensorAlignment quantizedTensorAlignmentOnActivations);
+    void setQuantizedTensorAlignmentOnWeights(const QuantizedTensorAlignment quantizedTensorAlignmentOnWeights);
+
+    void setQuantizationIntervalAsymmetryThreshold(const float value);
+    void setZeroThreshold(const float value);
+    void setMinQuantizationLevels(const size_t levels);
+
+    const std::vector<element::Type>& getPrecisionsOnActivations() const;
+    const std::vector<element::Type>& getPrecisionsOnWeights() const;
+
+    virtual bool canBeTransformed(const TransformationContext& context, std::shared_ptr<Node> layer) const;
+
+    bool canSubtractBeHandled(const std::shared_ptr<Node>& op, const size_t parentIndex = 0ul) const;
+
+    bool canSubtractBeHandled(const std::shared_ptr<Node>& op, const FakeQuantizeDequantization& dequantization) const;
+
+    PrecisionDetails getPrecisionDetails(const QuantizationDetails& quantizationDetails) const;
+
+    // return true if operation can be quantized and false otherwise
+    // for example: if convolution operation weights are not quantized, then isQuantize returns false and true otherwise
+    // note: dequantization operations on activations are absent during method execution
+    virtual bool isQuantized(std::shared_ptr<Node> layer) const noexcept;
+
+    // return true if operation can be preserved for precision
+    // note: dequantization operations on activations are absent during method execution
+    virtual bool isPrecisionPreserved(std::shared_ptr<Node> layer) const noexcept = 0;
+
+    DataPrecision getDataPrecision(
+            std::shared_ptr<Node> layer,
+            const QuantizationDetails& quantizationDetails,
+            const bool onWeights) const;
+
+    void fillAvailablePrecisions(std::shared_ptr<Node> layer, std::vector<element::Type>& availablePrecisions) const;
+
+    std::vector<std::shared_ptr<Node>> getChildrenRecursivelyExceptPrecisionPreserved(const std::shared_ptr<Node>& op) const noexcept;
+
+protected:
+#ifdef LPT_PRINT_DEQUANTIZATION_INFO
+    static void printDequantizationInfo(const std::shared_ptr<Node>& layer);
+    static void printDequantizationInfo(const DataPrecision& dataPrecision);
+    static void printDequantizationValues(
+        const std::vector<float>& dequantizationScales,
+        const std::vector<float>& dequantizationShifts);
+#endif
+
+    bool updatePrecisions;
+    QuantizedTensorAlignment quantizedTensorAlignmentOnActivations;
+    QuantizedTensorAlignment quantizedTensorAlignmentOnWeights;
+    bool supportAsymmetricQuantization;
+    std::vector<element::Type> precisionsOnActivations;
+    std::vector<element::Type> precisionsOnWeights;
```

Comment:
this is a duplicates of `Params` class content. Why you don't just use `params` structure here?